OpenRA.Mods.Common/Widgets/Logic/Ingame/MusicControllerLogic.cs

```diff
@@ -9,37 +9,59 @@
  */
 #endregion
 
+using System.Collections.Generic;
 using System.Linq;
 using OpenRA.GameRules;
 using OpenRA.Graphics;
+using OpenRA.Mods.Common.Lint;
 using OpenRA.Mods.Common.Traits;
 using OpenRA.Widgets;
 
 namespace OpenRA.Mods.Common.Widgets.Logic.Ingame
 {
+	[ChromeLogicArgsHotkeys("StopMusicKey", "PauseMusicKey", "PrevMusicKey", "NextMusicKey")]
 	public class MusicControllerLogic : ChromeLogic
-    {
-		MusicPlaylist musicPlaylist;
+	{
+		readonly MusicPlaylist musicPlaylist;
 
 		[ObjectCreator.UseCtor]
-		public MusicControllerLogic(Widget widget, World world, WorldRenderer worldRenderer)
+		public MusicControllerLogic(Widget widget, World world, Dictionary<string, MiniYaml> logicArgs)
 		{
 			musicPlaylist = world.WorldActor.Trait<MusicPlaylist>();
 
+			var ks = Game.Settings.Keys;
+			MiniYaml yaml;
+
+			var stopKey = new NamedHotkey();
+			if (logicArgs.TryGetValue("StopMusicKey", out yaml))
+				stopKey = new NamedHotkey(yaml.Value, ks);
+
+			var pauseKey = new NamedHotkey();
+			if (logicArgs.TryGetValue("PauseMusicKey", out yaml))
+				pauseKey = new NamedHotkey(yaml.Value, ks);
+
+			var prevKey = new NamedHotkey();
+			if (logicArgs.TryGetValue("PrevMusicKey", out yaml))
+				prevKey = new NamedHotkey(yaml.Value, ks);
+
+			var nextKey = new NamedHotkey();
+			if (logicArgs.TryGetValue("NextMusicKey", out yaml))
+				nextKey = new NamedHotkey(yaml.Value, ks);
+
 			var keyhandler = widget.Get<LogicKeyListenerWidget>("MUSICCONTROLLER_KEYHANDLER");
 			keyhandler.OnKeyPress = e =>
 			{
 				if (e.Event == KeyInputEvent.Down)
 				{
 					var key = Hotkey.FromKeyInput(e);
 
-					if (key == Game.Settings.Keys.NextTrack)
+					if (key == nextKey.GetValue())
 						musicPlaylist.Play(musicPlaylist.GetNextSong());
-					else if (key == Game.Settings.Keys.PreviousTrack)
+					else if (key == prevKey.GetValue())
 						musicPlaylist.Play(musicPlaylist.GetPrevSong());
-					else if (key == Game.Settings.Keys.StopMusic)
+					else if (key == stopKey.GetValue())
 						StopMusic();
-					else if (key == Game.Settings.Keys.PauseMusic)
+					else if (key == pauseKey.GetValue())
```

Comment:
only side node to ignore: this might be now only place with different/original order.